traveler-ios-ui/TravelerKitUI/ViewControllers/OrderDetailViewController.swift

```diff
@@ -53,10 +81,41 @@ class OrderDetailViewController: UITableViewController {
     }
 
     @IBAction func didCancelOrder(_ sender: Any) {
-        
+        if order?.status != OrderStatus.cancelled {
+            ProgressHUD.show()
+            Traveler.fetchCancellationQuote(order: order!, delegate: self)
+        }
     }
 
     @IBAction func didRequestTickets(_ sender: Any) {
 
     }
 }
+
+extension OrderDetailViewController: CancelOrderViewControllerDelegate {
+    func orderCancelSucceed(with order: Order) {
+        self.order = order
+        reload(with: self.order?.status)
```

Comment:
This has to report back to the list view controller..This is why I would normally opt for notifications instead of delegates